tendermint/src/lite_impl/signed_header.rs

```diff
@@ -63,6 +63,23 @@ impl lite::Commit for block::signed_header::SignedHeader {
         // https://github.com/interchainio/tendermint-rs/pull/143/commits/0a30022fa47e909e6c7b20417dd178c8a3b84838#r374958528
         // should go here or somewhere else
 
+        // make sure each vote is for the correct header
+        for precommit_opt in self.commit.precommits.iter() {
+            match precommit_opt {
+                Some(precommit) => {
+                    if let Some(header_hash) = precommit.header_hash() {
+                        if header_hash != self.header_hash() {
+                            fail!(
+                                lite::error::Kind::ImplementationSpecific,
+                                "vote from validator {} is not for the current header",
+                                precommit.validator_address
```

Comment:
Should we also include both hashes (header_hash, precommit.header_hash()) in the error string? At least in other similar places we do so and it might help debugging.

Reply:
Oh yes! Thanks, I'll update the code